### PR TITLE
Add first (and incomplete) draft for the Java SDK spec

### PR DESCRIPTION
**Pull request by JonathanGiles**

### CODE REVIEW

**Comment by Petermarcu** [comment]
Do we want this in this repo or the java one? I was thinking the java one

**Comment by JonathanGiles** [comment]
I'm fine with wherever - this was where we discussed today that it should go, but direct me to wherever you want it to go, and I'll resubmit (I note the .net markdown is in this repo already).

**Comment by JonathanGiles** [comment]
As discussed in the arch board meeting today, I will now take the liberty of merging this pull request myself, but I encourage all reviewers to leave any thoughts that they may have so that they may be worked into future revisions. Thanks.

File: docs/design/AzureSDKDesignGuidelines-java.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+This specification is lacking detail in a number of areas. You can help - please provide pull requests to fill in important gaps! In general the organization of the specification could also be improved. Suggestions welcome!
+
+## 2 Scope
+
+Any SDK published to Maven Central under the `com.azure` group ID scope must follow these conventions. For other SDKs, including Azure internal SDKs, these guidelines are merely recommended.
+
+## 3 Terminology
+
+- **SDK**: Software Development Kit. This refers to a release of Java class, JavaDoc, and source code jar files published to Maven Central, typically as an official GA release, but it may also be used to refer to a snapshot release. These releases are performed exclusively by the Azure SDK engineering systems team.
```
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply docs/design/AzureSDKDesignGuidelines-java.md]: This change sounds good. I have redefined the terms in my latest draft as such:

- **Azure SDK Component**: An Azure SDK component (or 'component') represents the software (and associated tools, documentation, samples, etc) that exists to support a single Azure service. Each Azure SDK component will be published separately to Maven Central under a common `com.azure` groupId, but with a unique `artifactId`. These releases are performed exclusively by the Azure SDK engineering systems team. Developers will be able to consume and use each component separately as necessary to solve their use case.
- **SDK**: Software Development Kit. This refers to the entire Azure SDK for Java, itself broken up into numerous Azure SDK Components as defined above.

File: docs/design/AzureSDKDesignGuidelines-java.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+Each dependency we add to an SDK increases its weight, and so it should be done with caution and careful consideration.
+
+Because each SDK pom.xml file refers upwards to a parent pom for all Azure SDKs, there is a centralized location for dependencies to be detailed. In Maven, when a parent pom.xml specifies a list of dependencies, this does not imply that all such dependencies are imported into the project. In this way, the parent pom.xml dependencies can be thought of as nothing more than a mapping from `groupId:artifactId` to `version`. Because this feature exists, all pom.xml files for all SDKs should rarely specify a specific version for a dependency - this should instead be inherited from the parent pom. In doing so, we can avoid many of the pitfalls of dependency management, commonly referred to in Java parlance as 'classpath hell'.
+
+> TODO document how to add dependencies to the parent pom, how to update versions, etc
+
+## 11 Coding Conventions
+
+Library APIs must be idiomatic and mst follow best practices.
```
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply docs/design/AzureSDKDesignGuidelines-java.md]: Thanks

**Pull request merged**